### DISCUSSION TITLE
Use handlr for mime detection

### DISCUSSION THREAD

**Discussion by raffaelemancuso** [sxyazi/yazi#1016] [1. Q&A]
The `file` command detects my neovim configuration files as `text/plain` at best and `application/javascript` at worst.

While the correct mime is `text/x-lua`.

I use [handlr-regex](https://github.com/Anomalocaridid/handlr-regex) as my file opener.

It can detect the mime type of my LUA files correctly.

How can I instruct yazi to use handlr for mimetype detection?

**Comment by sxyazi** [comment]
Create a `mime` preloader plugin based on `handlr` instead of `file(1)` to get the mime-type of the file. i.e. copy [https://github.com/sxyazi/yazi/blob/main/yazi-plugin/preset/plugins/mime.lua](https://github.com/sxyazi/yazi/blob/main/yazi-plugin/preset/plugins/mime.lua) to `~/.config/yazi/plugins/mime.yazi/init.lua`, then change the `file` command in it to `handlr`. Here's an example: [https://github.com/DreamMaoMao/mime.yazi](https://github.com/DreamMaoMao/mime.yazi)
